### PR TITLE
review fix: contract CtAnnotation and array of values

### PR DESCRIPTION
**Pull request by surli**
This PR intends to fix the contract defined after discussion in #1928

### CODE REVIEW

File: src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java [hunk 1]
```diff
@@ -357,6 +358,31 @@ private Object getDefaultValue(String fieldName) {
...
+			if (type != null) {
+				CtMethod method = type.getMethod(key);
+				if (method != null) {
+					CtTypeReference returnType = method.getType();
+					if (returnType instanceof CtArrayTypeReference) {
+						CtNewArray newArray = getFactory().Core().createNewArray();
+						CtArrayTypeReference typeReference2 = this.getFactory().createArrayTypeReference();
+						typeReference2.setComponentType(ctExpression.getType().clone());
+						newArray.setType(typeReference);
+						newArray.addElement(ctExpression);
```
Comment (pvojtechovsky) [inline src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java]: `newArray.addElement(ctExpression.clone());`
clone is missing

File: src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java [hunk 2]
```diff
@@ -357,6 +358,31 @@ private Object getDefaultValue(String fieldName) {
 		return (T) this.elementValues.get(key);
 	}
 
+	@Override
+	public <T extends CtExpression> T getTypedValue(String key) {
+		CtExpression ctExpression = this.getValue(key);
+
+		if (ctExpression instanceof CtLiteral) {
+			CtTypeReference typeReference = this.getAnnotationType();
+			CtType type = typeReference.getDeclaration();
```
Comment (monperrus) [inline src/main/java/spoon/support/reflect/declaration/CtAnnotationImpl.java]: use getTypeDeclaration?

**Comment by monperrus** [comment]
I have a doubt about the name `getTypedValue` because all values are already typed.

Does this method always return a CtNewArray?

**Comment by surli** [comment]
> Does this method always return a CtNewArray?

only if it respects some conditions (it's the only CtLiteral value and the annotation should return an array) else it returns the same value as `getValue`.

**Comment by monperrus** [comment]
what about getWrappedValue?

**Comment by monperrus** [comment]
Thanks Simon!

**Pull request merged**